### PR TITLE
CORE-19693: Throw 503 when Kafka is slow

### PR DESCRIPTION
**Pull request by yift-r3**

### CODE REVIEW

File: components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MGMResourceClientImpl.kt [hunk 1]
```diff
@@ -849,28 +852,40 @@ class MGMResourceClientImpl @Activate constructor(
...
-
-                return response.response as RESPONSE
+            logger.debug { "Sending request: $this" }
+            val response = try {
+                rpcSender.sendRequest(this).getOrThrow(TIMEOUT)
             } catch (e: Exception) {
-                throw CordaRuntimeException(
-                    "Failed to send request and receive response for MGM RPC operation. " + e.message,
-                    e
+                throw ServiceNotReadyException(e)
```
Comment (williamvigorr3) [inline components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MGMResourceClientImpl.kt:861]: I think we should catch CordaRPCAPIPartitionException specifically (we shouldn't return a 503 if something internally went wrong, so we should additionally keep the old try catch block)?.

Or that and a TimeoutException.

Comment (yift-r3) [reply components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MGMResourceClientImpl.kt:861]: We need both.

Comment (williamvigorr3) [reply components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MGMResourceClientImpl.kt:861]: I still think catching `Exception` is too generic.

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Pull request merged**